backends/disk.go

```diff
@@ -42,16 +47,64 @@ func NewDiskBackend(config Config) (*DiskBackend, error) {
 	return &DiskBackend{safePath}, nil
 }
 
-func (db *DiskBackend) Load() ([]byte, error) {
+func (db *DiskBackend) Load() (crypto.Config, []byte, error) {
+	var config crypto.Config
+	var ciphertext []byte
+
 	if _, err := os.Stat(db.path); os.IsNotExist(err) {
-		return nil, &errors.SafeNotFound{}
+		return config, nil, &errors.SafeNotFound{}
+	}
+
+	file, err := os.Open(db.path)
+	if err != nil {
+		return config, nil, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for i := 0; scanner.Scan(); i++ {
+		b := scanner.Bytes()
+		if i == 0 {
+			// First line, should be either config or, for backwards-compatibility, ciphertext
+			err := json.Unmarshal(b, &config)
+			if err == nil {
+				// Unmarshalling succeeded -> Current line is config
+				// Throw it away and continue with the next line.
+				continue
+			}
+		}
+		// Base64-decode ciphertext
+		ciphertextLen := base64.StdEncoding.DecodedLen(len(b))
+		ciphertext = make([]byte, ciphertextLen)
+		n, err := base64.StdEncoding.Decode(ciphertext, b)
+		if err != nil {
+			return config, nil, err
+		}
+		if n > ciphertextLen {
+			return config, nil, &errors.SafeCorrupt{}
+		}
+		// Remove NULL bytes
+		ciphertext = bytes.Trim(ciphertext, "\x00")
+		break
+	}
+	if err := scanner.Err(); err != nil {
+		return config, nil, err
 	}
 
-	return ioutil.ReadFile(db.path)
+	return config, ciphertext, nil
 }
 
-func (db *DiskBackend) Save(data []byte) error {
-	return ioutil.WriteFile(db.path, data, defaultSafeFileMode)
+func (db *DiskBackend) Save(config crypto.Config, ciphertext []byte) error {
+	cnf, err := json.Marshal(config)
+	if err != nil {
+		return err
+	}
+	// Base64-encode ciphertext
+	ciphertextB64 := make([]byte, base64.StdEncoding.EncodedLen(len(ciphertext)))
+	base64.StdEncoding.Encode(ciphertextB64, ciphertext)
+	data := fmt.Sprintf("%s\n%s", cnf, ciphertextB64)
```

Comment:
This will work for the `Save` function, but the client still needs to differentiate between these two cases:

1. The backward-compatible mode, where `pick.safe` includes _just_ the message returned from OpenPGP
2. The new mode, which also stores the config

The proposed JSON format is much better. Will update the PR.